usr/lib/xtable/libxtable.c

```diff
@@ -0,0 +1,218 @@
+/*
+* SPDX-FileCopyrightText: (c) 2024 Mohammad Shehar Yaar Tausif <sheharyaar48@gmail.com>
+* SPDX-License-Identifier: MIT OR GPL-2.0-only
+*/
+
+#include <lauxlib.h>
+#include <lua.h>
+#include <lualib.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <xtables.h>
+
+#include "luaxtable.h"
+
+#ifndef LUAXTABLE_MODULE
+#define LUAXTABLE_MODULE "test"
+#endif
+
+static lua_State *L = NULL;
+const char *libxtable_match_key = "libxtable_match";
+const char *libxtable_target_key = "libxtable_target";
+
+int luaopen_libxtable(lua_State *L);
+
+// Function to initialize Lua state
+static int init_lua(void)
+{
+	L = luaL_newstate();
+	if (L == NULL) {
+		fprintf(stderr, "Failed to create Lua state\n");
+		return 0;
+	}
+	luaL_openlibs(L);
+	luaL_requiref(L, "libxtable", luaopen_libxtable, 1);
+	return 1;
+}
+
+// Function to load Lua script
+static int load_lua_script(const char *filename)
+{
+	if (luaL_dofile(L, filename) != LUA_OK) {
+		fprintf(stderr, "Failed to load Lua script: %s\n", lua_tostring(L, -1));
+		return 0;
+	}
+	return 1;
+}
+
+// Helper function to call Lua functions
+static int libxtable_invoke(const char *func_name, int nargs, int nresults)
+{
+	int ret = 0;
+	int base = lua_gettop(L) - nargs;
+	if (lua_getfield(L, -1, func_name) != LUA_TFUNCTION) {
+		fprintf(stderr, "Function %s not found\n", func_name);
+		return 0;
+	}
+
+	lua_insert(L, base + 1); /* func */
+	if (lua_pcall(L, nargs, nresults, 0) != LUA_OK) {
+		fprintf(stderr, "Failed to call Lua function %s: %s\n", func_name,
+				lua_tostring(L, -1));
+		lua_settop(L, base - 1);
+		return 0;
+	}
+
+	if (nresults == 1)
+		ret = lua_toboolean(L, -1);
+
+	lua_settop(L, base - 1);
+	return ret;
+}
+
+static inline int libxtable_run(lua_State *L, const char *func_name, const char *key, int nargs, int nresults)
+{
+	lua_rawgetp(L, LUA_REGISTRYINDEX, key);
+	return libxtable_invoke(func_name, nargs, nresults);
```

Comment:
I don't think we need this indirection here.. I would just move the body of this function inhere..